pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexCRUDViolationRule.java

```diff
@@ -657,12 +657,22 @@ private boolean validateCRUDCheckPresent(final ApexNode<?> node, final Object da
         boolean isImproperDMLCheck = !isProperESAPICheckForDML(typeCheck, crudMethod)
                 && !isProperAuthPatternBasedCheckForDML(typeCheck, crudMethod);
         boolean noSecurityEnforced = !isWithSecurityEnforced(node);
+        boolean userMode = isWithUserMode(node);
+        boolean systemMode = isWithSystemMode(node);
         if (missingKey) {
             //if condition returns true, add violation, otherwise return.
             if (isImproperDMLCheck && noSecurityEnforced) {
                 addViolation(data, node);
                 return true;
             }
+            if (isImproperDMLCheck && userMode) {
```

Comment:
```suggestion
            if (isImproperDMLCheck && !userMode && !systemMode) {
```

If I understand it correctly, using "User Mode" is good and should not be flagged. So, only if it is neither user mode nor system mode specified, then Apex uses the defaults, which is elevated system mode, which is bad.

Reply:
yeah you are right